### PR TITLE
Transient Reentrancy Guard

### PR DESCRIPTION
**Pull request by syntrust**
Upgrade solidity to v0.8.28 as it fully supports transient state variables of value types. Thus we can get rid of the lengthy code of "@openzeppelin/contracts/utils/ReentrancyGuardTransient.sol" and its dependencies introduced in https://github.com/ethstorage/storage-contracts-v1/pull/108.

Reference:
https://docs.soliditylang.org/en/v0.8.28/contracts.html#transient-storage

### CODE REVIEW

**Review by qzhodl — APPROVED** [review]
LGTM

File: package.json [hunk 1]
```diff
@@ -11,23 +11,15 @@
     "@nomiclabs/hardhat-web3": "^2.0.0",
     "@openzeppelin/contracts": "^5.0.1",
     "@openzeppelin/contracts-upgradeable": "^5.0.1",
-    "arb-ts": "^1.0.2",
     "chai": "^4.3.4",
     "dotenv": "^10.0.0",
```
Comment (qizhou) [inline package.json:15]: Why some of the packages such as prettier is removed?

**Review by qizhou — COMMENTED** [review]
I suggest no mixing the change of the code with coding style reformating.

File: package.json (hunk 1 above)
Comment (syntrust) [reply package.json]: Just some cleanup work without breaking tests, plus we agreed to use forge as the formatter. Maybe it's more safe for everyone to revert it back.

**Review by iteyelmp — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (qzhodl) [reply package.json]: We can change to prettier to forge in prettier:check & prettier:fix of the package.json

**Comment by qzhodl** [comment]
> I suggest no mixing the change of the code with coding style reformating.

Yes, I think it is a good practice that we need to follow

**Pull request merged**